### ISSUE TITLE
Allow to send custom message in `checkError`

### ISSUE THREAD

**Issue by dklimpel** [marmelab/react-admin#6547]
**Is your feature request related to a problem? Please describe.**
In `checkAuth` it is possible to send a custom message `message: string`
https://github.com/marmelab/react-admin/blob/d33a4ee57e1e6347c9f9ad435b197c00af2d6e3a/docs/Authentication.md#L609
https://github.com/marmelab/react-admin/blob/d33a4ee57e1e6347c9f9ad435b197c00af2d6e3a/packages/ra-core/src/auth/useCheckAuth.ts#L67-L70

I would like prefer the same in `checkError`. Than is it possible to show in dependency of the error the matching message.
At the moment the message is fixed and you can only disable it.
https://github.com/marmelab/react-admin/blob/d33a4ee57e1e6347c9f9ad435b197c00af2d6e3a/docs/Authentication.md#L608
https://github.com/marmelab/react-admin/blob/d33a4ee57e1e6347c9f9ad435b197c00af2d6e3a/packages/ra-core/src/auth/useLogoutIfAccessDenied.ts#L73-L82

**Describe the solution you'd like**
I would like to send custom messages in `checkError` like:
```js
  checkError: ({ status }) => {
    if (status === 401) {
      return Promise.reject({ message: "Message 401" });
    } else if (status === 403) {
      return Promise.reject({ message: "Message 403" });
    }
    return Promise.resolve();
  },
```

**Comment by fzaninotto** [comment]
Nice idea! Feel free to open a PR against the `next` branch to add the feature to the core.

**Comment by dklimpel** [comment]
Ok, I think it needs a few weeks.

**Comment by fzaninotto** [comment]
Fixed by #6619
